src/main/java/school/lemon/changerequest/java/io/pr1/task3/CountEngineImp.java

```diff
@@ -0,0 +1,163 @@
+        if (!file.exists()) throw new FileNotFoundException();
+        try {
+            fis = new FileInputStream(file);
+            bis = new BufferedInputStream(fis);
+            while (bis.available() > 0) {
+                if ((char) bis.read() == c) result++;
+            }
+        } catch (IOException n) {
+        }
+        return result;
+    }
+
+    @Override
+    public long countCharacters(String filename) throws FileNotFoundException {
+        long result = 0;
+        File file = new File(filename);
+        BufferedInputStream bis;
+        FileInputStream fis;
+        if (!file.exists()) throw new FileNotFoundException();
+        try {
+            fis = new FileInputStream(file);
+            bis = new BufferedInputStream(fis);
+            while (bis.available() > 0) {
+                bis.read();
+                result++;
+            }
+        } catch (IOException n) {
+        }
+        return result;
+    }
+
+    @Override
+    public long countWords(String filename) throws FileNotFoundException {
+        File file = new File(filename);
+        if (!file.exists()) {
+            throw new FileNotFoundException();
+        }
+        char c;
+        int words = 0;
+        int charForWords = 0;
+        int spaces = 0;
+        boolean wordWasCounted = false;
+        boolean firstWord = true;
+        Pattern charsP = Pattern.compile("[a-zA-Z0-9]");
+        Pattern spacesP = Pattern.compile("[\n\\s]");
+        Matcher charsM;
+        Matcher spacesM;
+        try {
+            BufferedInputStream bufferedInputStream = new BufferedInputStream(new FileInputStream(file));
+            c = (char) bufferedInputStream.read();
+            CharSequence str = Character.toString(c);
+            while (bufferedInputStream.available() > 0) {
+                charsM = charsP.matcher(str);
+                spacesM = spacesP.matcher(str);
+                if (spacesM.matches()) {
+                        spaces++;
+                        wordWasCounted = false;
+                        charForWords = 0;
+                } else if ((charsM.matches() && (spaces > 0) || firstWord)) {
+                    firstWord = false;
+                    charForWords++;
+                    if (charForWords == 1 && !wordWasCounted) {
+                        words++;
+                        wordWasCounted = true;
+                    }
+                }
+                c = (char) bufferedInputStream.read();
+                str = Character.toString(c);
+            }
+        } catch (IOException n) {
+        }
+        return words;
+    }
+
+    @Override
+    public long countParagraphs(String filename) throws FileNotFoundException {
+        File file = new File(filename);
+        int paragraphs = 0;
+        boolean firstParagraph = true;
+        if (!file.exists()) {
+            throw new FileNotFoundException();
+        }
+        BufferedReader br = new BufferedReader(new FileReader(file));
```

Comment:
You do not need to count words in line here. You need read whole line and check whether it's empty line. Then, if there more than 2 empty lines in a row - increase paragraphs count.